File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/domain/type/IFizzBuzzType.java
package fizzbuzz.domain.type;

import fizzbuzz.domain.model.FizzBuzzValue;

abstract class IFizzBuzzType {
  public static final String FIZZ_BUZZ = "FizzBuzz";
  public static final String FIZZ = "Fizz";
  public static final String BUZZ = "Buzz";

  abstract FizzBuzzValue generate(int number);

}

File: dev/20190822/k2works/src/Application/Service/IFizzBuzzType.ts
import { FizzBuzzValue } from "../../Domain/Model/FizzBuzzValue";

export abstract class IFizzBuzzType {
  public static readonly FIZZ: string = "Fizz";
  public static readonly BUZZ: string = "Buzz";
  public static readonly FIZZ_BUZZ: string = "FizzBuzz";
  public abstract generate(input: number): FizzBuzzValue;
}
